src/proofs/range_proof_wip.rs

```diff
@@ -0,0 +1,571 @@
+#![allow(non_snake_case)]
+
+/*
+
+Copyright 2018 by Kzen Networks
+
+This file is part of bulletproof library
+(https://github.com/KZen-networks/bulletproof)
+
+bulletproof is free software: you can redistribute
+it and/or modify it under the terms of the GNU General Public
+License as published by the Free Software Foundation, either
+version 3 of the License, or (at your option) any later version.
+
+@license GPL-3.0+ <https://github.com/KZen-networks/bulletproof/blob/master/LICENSE>
+*/
+
+// based on the paper: https://eprint.iacr.org/2020/735.pdf
+//
+// Bulletproofs (https://eprint.iacr.org/2017/1066) uses the inner product argument.
+// Bulletproofs+ (https://eprint.iacr.org/2020/735.pdf) uses the weighted inner product argument
+// which reduces the overall prover communication by ~15%
+// 
+use curv::arithmetic::traits::{Converter, Modulo};
+use curv::cryptographic_primitives::hashing::hash_sha256::HSha256;
+use curv::cryptographic_primitives::hashing::traits::*;
+use curv::elliptic::curves::traits::*;
+use curv::BigInt;
+use curv::{FE, GE};
+use itertools::iterate;
+use proofs::weighted_inner_product::WeightedInnerProdArg;
+use std::ops::{Shl, Shr};
+use Errors::{self, RangeProofError};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct RangeProofWIP {
+    A: GE,
+    weighted_inner_product_proof: WeightedInnerProdArg,
+}
+
+impl RangeProofWIP {
+    pub fn prove(
+        g_vec: &[GE],
+        h_vec: &[GE],
+        G: &GE,
+        H: &GE,
+        mut secret: Vec<FE>,
+        blinding: &[FE],
+        bit_length: usize,
+    ) -> RangeProofWIP {
+
+        let num_of_proofs = secret.len();
+        //num of proofs times bit length
+        let nm = num_of_proofs * bit_length;
+        
+        let g_vec = g_vec.to_vec();
```

Comment:
no assertion about equality of g_vec and h_vec?

Reply:
Done!